### PR TITLE
fix: handling of aliases improved

### PR DESCRIPTION
**Pull request by drktfl**
Issue: #492

This PR improves the handling of aliases in the annotation converter.

As an example, the annotation converter now resolves action references with aliases:

```xml
<edmx:Reference Uri="">
    <edmx:Include Alias="MyServiceAlias" Namespace="sap.fe.test.JestService"/>
</edmx:Reference>
...
<Action Name="doSomething" IsBound="true">
    <Parameter Name="in" Type="sap.fe.test.JestService.Entities"/>
</Action>
...
<Record Type="MyUI.DataFieldForAction">
    <PropertyValue Property="Action" String="MyServiceAlias.doSomething"/>
</Record>
```

### CODE REVIEW

**Review by nlunets — COMMENTED** [review]
That looks interesting :) 
In this context maybe a changeset would be required.
I'm now wondering if we should also push to transform all kind of annotaiton where we know we have actions (introducing the target) or at least provide a `resolveAction` method.

By the way can you include tests for all kind of action name resolving ? https://github.com/SAP/odata-vocabularies/blob/main/vocabularies/UI.md#ActionName i think you're missing some (simple name and qualified name without entity type specifiier)

**Comment by drktfl** [comment]
This should now improve the alias handling in general.

It does not address `resolveAction` or conversion of action targets other than the special handling for DataFieldForAction that is already in place. I feel this is something for a follow-up change to keep this one manageable ;-)

File: packages/annotation-converter/src/converter.ts [hunk 1]
```diff
@@ -572,6 +573,18 @@
     );
 }
 
+function splitActionFQN(actionFQN: FullyQualifiedName | undefined) {
+    // this.is.the.action(on.this.type) --> action: 'this.is.the.action', overload: 'on.this.type'
+    // this.is.the.action()             --> action: 'this.is.the.action', overload: undefined
+    // this.is.the.action               --> action: 'this.is.the.action', overload: undefined
+
+    const actionAndOverload = actionFQN?.match(/(?<action>[^()]+)(?:\((?<overload>.*)\))?/);
```
[the remark this answers is not in the document]
Comment (drktfl) [reply packages/annotation-converter/src/converter.ts:581]: I think this regular expression can be replaced. Will push another change.

**Comment by drktfl** [comment]
Fixes #492

**Review by nlunets — APPROVED** [review]
Looks good to me !
The tests also show all the things we want to do.
And let's look into having a resolveAction independently

**Pull request merged**